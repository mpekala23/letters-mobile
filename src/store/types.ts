import { UserState } from "./User/UserTypes";
import { NotifState } from "./Notif/NotifTypes";
import { ContactState } from "./Contact/ContactTypes";

export interface AppState {
  user: UserState;
  notif: NotifState;
  contact: ContactState;
}
