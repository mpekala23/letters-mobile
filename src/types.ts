// TODO: make this typing better
// probably want a specific type for state abbrevs
export interface Facility {
  name: string;
  type: PrisonTypes;
  address: string;
  city: string;
  state: string;
  postal: string;
  phone: string;
}

export enum ProfilePicTypes {
  Topbar = 'Topbar',
  Contact = 'Contact',
  SingleContact = 'SingleContact',
}

export enum Storage {
  RememberToken = 'Ameelio-Token',
}

export enum PrisonTypes {
  State = 'State',
  Federal = 'Federal',
  Jail = 'County',
}

export enum SupportFAQTypes {
  DeleteLetter = 'DeleteLetter',
  NotArrived = 'NotArrived',
  WrongReturnAddress = 'WrongReturnAddress',
  WrongMailingAddress = 'WrongMailingAddress',
  TrackingNumber = 'TrackingNumber',
  TrackingError = 'TrackingError',
  TalkToAmeelio = 'TalkToAmeelio',
}

export enum LetterStatus {
  Draft = 'Draft',
  Created = 'Created',
  Mailed = 'Mailed',
  InTransit = 'In Transit',
  InLocalArea = 'In Local Area',
  OutForDelivery = 'Out for Delivery',
  Delivered = 'Delivered',
}

export enum LetterTypes {
  Postcard = 'postcard',
  Letter = 'letter',
}

export enum DeliveryReportTypes {
  received = 'received',
  unsure = 'unsure',
  notYetReceived = 'notYetReceived',
  haveNotAsked = 'haveNotAsked',
  haveNotReceived = 'haveNotReceived',
}

export interface Letter {
  type: LetterTypes;
  status: LetterStatus;
  isDraft: boolean;
  recipientId: number;
  content: string;
  photo?: Photo;
  letterId?: number; // TODO: Once we have more info on this field and lob, use this more
  expectedDeliveryDate?: Date;
  dateCreated?: Date;
  trackingEvents?: LetterTrackingEvent[];
}

export interface LetterTrackingEvent {
  id: number;
  name: string;
  location: ZipcodeInfo;
  date: Date;
}

export type TopbarRouteAction = {
  enabled: boolean;
  text: string;
  action: () => void | Promise<void>;
  blocking?: boolean;
};

export interface ZipcodeInfo {
  zip: string;
  city: string;
  state: string;
  lat?: number;
  long?: number;
}

export interface Photo {
  type?: 'image' | string;
  width?: number;
  height?: number;
  uri: string;
}
