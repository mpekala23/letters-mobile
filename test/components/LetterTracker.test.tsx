import React from 'react';
import { LetterTracker } from '@components';
import { render, toJSON } from '@testing-library/react-native';
import { LetterStatus } from 'types';

jest.mock('moment', () => () => ({
  format: () => 'Jul 12',
}));

const setup = (propOverrides = {}) => {
  const props = {
    trackingEvent: {
      id: 1,
      name: 'Mailed',
      location: 'Zipcode',
      date: '2019-07-12T15:51:41.000Z',
      ...propOverrides,
    },
  };
  return {
    ...render(<LetterTracker {...props} />),
    props,
  };
};

describe('Letter Tracker component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should display letter status', () => {
    const { getByText } = setup();
    expect(getByText('Mailed')).toBeDefined();
  });

  it('should display zipcode', () => {
    const { getByText } = setup();
    expect(getByText('Zipcode')).toBeDefined();
  });

  it('should display formatted date', () => {
    const { getByTestId } = setup();
    expect(getByTestId('dateFormatted').props.children).toBe('Jul 12');
  });

  it('should display tracker color for status Mailed', () => {
    const { getByTestId } = setup({ name: LetterStatus.Mailed });
    expect(getByTestId('trackerCircle').props.style.backgroundColor).toBe(
      '#A8C1E4'
    );
  });

  it('should display tracker color for status InTransit', () => {
    const { getByTestId } = setup({ name: LetterStatus.InTransit });
    expect(getByTestId('trackerCircle').props.style.backgroundColor).toBe(
      '#8DA7CC'
    );
  });

  it('should display tracker color for status InLocalArea', () => {
    const { getByTestId } = setup({ name: LetterStatus.InLocalArea });
    expect(getByTestId('trackerCircle').props.style.backgroundColor).toBe(
      '#6D89B1'
    );
  });

  it('should display tracker color for status OutForDelivery', () => {
    const { getByTestId } = setup({ name: LetterStatus.OutForDelivery });
    expect(getByTestId('trackerCircle').props.style.backgroundColor).toBe(
      '#436697'
    );
  });
});
