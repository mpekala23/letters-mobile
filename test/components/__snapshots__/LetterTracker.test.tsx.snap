// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter Tracker component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 16,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F0FAF3",
            "borderRadius": 50,
            "height": 28,
            "marginLeft": 4,
            "marginRight": 24,
            "width": 28,
          }
        }
        testID="trackerCircle"
      />
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Bold",
              },
              Object {
                "color": "#282828",
                "fontSize": 16,
                "paddingBottom": 6,
              },
            ]
          }
        >
          Mailed
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Bold",
              },
              Object {
                "color": "#393939",
                "fontSize": 14,
                "paddingBottom": 4,
              },
            ]
          }
        >
          Zipcode
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "column",
            "marginLeft": "auto",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Regular",
              },
              Object {
                "color": "#393939",
                "fontSize": 14,
                "paddingBottom": 8,
              },
            ]
          }
          testID="dateFormatted"
        >
          Jul 12
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Regular",
              },
              Object {
                "color": "#393939",
                "fontSize": 14,
                "paddingBottom": 4,
              },
            ]
          }
        >
          Jul 12
        </Text>
      </View>
    </View>
  </View>
</View>
`;
